remote-api/src/main/java/org/alfresco/rest/api/model/Tag.java

```diff
@@ -101,36 +101,20 @@ public int compareTo(Tag o)
 	}
 
 	@Override
-	public int hashCode()
+	public boolean equals(Object o)
 	{
-		final int prime = 31;
-		int result = 1;
-		result = prime * result + ((nodeRef == null) ? 0 : nodeRef.hashCode());
-		return result;
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
+			return false;
+		Tag tag1 = (Tag) o;
+		return Objects.equals(nodeRef, tag1.nodeRef) && Objects.equals(tag, tag1.tag) && Objects.equals(count, tag1.count);

```

Comment:
Is that really the case? I think `NodeRef` should be enough as `count` field not always gets populated but I may be wrong. When comparing a tag without `count` with the same tag with`count` populated - they should still be equal, shouldn't they? Tag's `name` (aka `tag`) must be the same for the same node as it points to the same node property.
Unless there are some read/write inconsistencies nothing more than `NodeRef` (tag's `NodeRef` is the node reference of tag node) should be needed here to make sure two tags are equal, shoud there?

Reply:
We probably don't use `Tag` instances in hash-based structures (do we?) but if we did - then this `equals`/`hasCode` implementation could result in having unwanted duplicates.

Reply:
No, `Tag`s are used only with lists. Regarding `count` field - almost all the time it is `null`, so I assumed that it shouldn't be a huge deal even for hash-based structures.